tests/test_nhg.py

```diff
@@ -384,20 +799,891 @@ def asic_route_nhg_fvs(k):
+
+            if fvs["SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID"] == nhgid:
+                count += 1
+
+        assert count == 3
+
+        # create prefix route in APPL DB
+
+        rt_ps = swsscommon.ProducerStateTable(app_db.db_connection, "ROUTE_TABLE")
+
+        asic_routes_count = len(asic_db.get_keys(self.ASIC_RT_STR))
+
+        fvs = swsscommon.FieldValuePairs([("nexthop_group", "group1")])
+        rt_ps.set("2.2.2.0/24", fvs)
+
+        # check if route was propagated to ASIC DB
+
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_routes_count + 1)
+
+        k = self.get_route_id('2.2.2.0/24', dvs)
+
+        # assert the route points to next hop group
+        fvs = asic_db.get_entry(self.ASIC_RT_STR, k)
+        assert fvs["SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID"] == nhgid
+
+        # create label route in APPL DB pointing to the same next hop group
+
+        lrt_ps = swsscommon.ProducerStateTable(app_db.db_connection, "LABEL_ROUTE_TABLE")
+
+        asic_insegs_count = len(asic_db.get_keys(self.ASIC_INSEG_STR))
+
+        fvs = swsscommon.FieldValuePairs([("nexthop_group", "group1")])
+        lrt_ps.set("20", fvs)
+
+        # check if label route was propagated to ASIC DB
+
+        asic_db.wait_for_n_keys(self.ASIC_INSEG_STR, asic_insegs_count + 1)
+
+        k = self.get_inseg_id('20', dvs)
+        assert k is not None
+
+        # assert the route points to next hop group
+        fvs = asic_db.get_entry(self.ASIC_INSEG_STR, k)
+        assert fvs["SAI_INSEG_ENTRY_ATTR_NEXT_HOP_ID"] == nhgid
+
+        # bring links down one-by-one
+        for i in [0, 1, 2]:
+            self.flap_intf(i, 'down', dvs)
+
+            keys = asic_db.get_keys(self.ASIC_NHGM_STR)
+
+            assert len(keys) == (asic_nhgms_count + 2 - i)
+            assert bool(asic_db.get_entry(self.ASIC_NHG_STR, nhgid))
+
+        # bring links up one-by-one
+        for i in [0, 1, 2]:
+            self.flap_intf(i, 'up', dvs)
+
+            keys = asic_db.get_keys(self.ASIC_NHGM_STR)
+
+            assert len(keys) == (asic_nhgms_count + i + 1)
+
+            count = 0
+            for k in keys:
+                fvs = asic_db.get_entry(self.ASIC_NHGM_STR, k)
+                if fvs["SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_GROUP_ID"] == nhgid:
+                    count += 1
+            assert count == i + 1
+
+        # Bring an interface down
+        self.flap_intf(1, 'down', dvs)
+
+        # One group member will get deleted
+        asic_db.wait_for_n_keys(self.ASIC_NHGM_STR, 2)
+
+        # Remove route 2.2.2.0/24
+        rt_ps._del("2.2.2.0/24")
+
+        # Wait for route 2.2.2.0/24 to be removed
+        asic_db.wait_for_n_keys(self.ASIC_RT_STR, asic_routes_count)
+
+        # Remove label route 20
+        lrt_ps._del("20")
+
+        # Wait for route 20 to be removed
+        asic_db.wait_for_n_keys(self.ASIC_INSEG_STR, asic_insegs_count)
+
+        # Remove group1
+        nhg_ps._del("group1")
```

Comment:
Additional tests:

* Same NHOP group created by routeOrch and nhgOrch are independent of each other. Create same nhop group with nhgOrch and routeOrch, test they have different SAI_NHOP_ID.

Reply:
We already have this in `test_nhgorch_excp_group_cases()`'s `test_routeorch_nhgorch_interop()`.